core/src/main/scala/kafka/server/FetcherEventManager.scala

```diff
@@ -169,7 +168,12 @@ class FetcherEventManager(name: String,
           processor.process(fetcherEvent)
         }
       } catch {
-        case e: Exception => error(s"Uncaught error processing event $fetcherEvent", e)
+        case e@(_: FatalExitError |
+                _: OutOfMemoryError |
+                _: StackOverflowError) =>
+          throw e
+        case e: Throwable => error(s"Uncaught error processing event $fetcherEvent", e)
+
```

Comment:
Per Slack discussion with Sean, there are lazy-class-loading Errors that this would catch and ignore, which is potentially a serious regression (though if we'd ever seen any of those, they'd have the same deadlock behavior as the InternalError, so _my_ gut feeling is it's a super-rare event, possibly completely negligible).  In any case, it's probably safer to make the minimal change, i.e., catch/log only InternalError and let the others go through.

Reply:
made the change